src/qtquickcomponents/sliderruler.cpp

```diff
@@ -0,0 +1,140 @@
+#include "sliderruler.h"
+
+#include <QPainter>
+#include <QPalette>
+#include <QFontMetrics>
+
+#include "stylemanager.h"
+
+SliderRuler::SliderRuler(QQuickItem *parent)
+    : QQuickPaintedItem(parent)
+    , m_count(-1)
+    , m_showText(true)
+    , m_controlColor("#90CAF9")
+{
+}
+
+void SliderRuler::setCount(int count)
+{
+    if (count != m_count) {
+        m_count = count;
+        emit countChanged(m_count);
+    }
+}
+
+int SliderRuler::count() const
+{
+    return m_count;
+}
+
+void SliderRuler::setShowText(bool showText)
+{
+    if (showText != m_showText) {
+        m_showText = showText;
+        emit showTextChanged(m_showText);
+    }
+}
+
+bool SliderRuler::showText() const
+{
+    return m_showText;
+}
+
+void SliderRuler::setTo(double to)
+{
+    if (!qFuzzyCompare(to, m_to)) {
+        m_to = to;
+        emit toChanged(m_to);
+    }
+}
+
+double SliderRuler::to() const
+{
+    return m_to;
+}
+
+void SliderRuler::setFrom(double from)
+{
+    if (!qFuzzyCompare(from, m_from)) {
+        m_from = from;
+        emit fromChanged(m_from);
+    }
+}
+
+double SliderRuler::from() const
+{
+    return m_from;
+}
+
+QPointF SliderRuler::beginLine() const
+{
+    return m_beginLine;
+}
+
+QPointF SliderRuler::endLine() const
+{
+    return m_endLine;
+}
+
+void SliderRuler::geometryChanged(const QRectF &newGeometry, const QRectF &oldGeometry)
+{
+    Q_UNUSED(oldGeometry)
+
+    const QRectF rect = newGeometry.adjusted(1, 1, -1, -1);
+    const auto height = rect.height();
+    const auto middlePoint = height / 2;
+
+    QFontMetrics fMetrics(qApp->font());
+    auto beginSize = fMetrics.horizontalAdvance(QString::number(qRound(m_from)));
+    auto endSize = fMetrics.horizontalAdvance(QString::number(qRound(m_to)));
+
+    const auto width = rect.width() - (beginSize / 2) - (endSize / 2);
+
+    m_beginLine = QPointF(rect.x() + beginSize / 2, middlePoint);
+    m_endLine = QPointF(m_beginLine.x() + width, middlePoint);
+    emit beginLineChanged(m_beginLine);
+    emit endLineChanged(m_endLine);
+    update();
+}
+
+void SliderRuler::paint(QPainter *painter)
+{
+    painter->save();
+
+    QFontMetrics fMetrics(qApp->font());
+
+    const QRectF rect = boundingRect().adjusted(1, 1, -1, -1);
+    const auto height = rect.height();
+    const auto middlePoint = height / 2;
+
+
+    const auto width = m_endLine.x() - m_beginLine.x();
+    const int tickCount = m_count != -1 ? m_count
+                     : m_to - m_from > 50 ? 50
+                     : static_cast<int>(m_to - m_from);
+
+    painter->setPen(m_controlColor);
+    painter->drawLine(m_beginLine, m_endLine);
+
+    constexpr int bitgTickSize = 10;
```

Comment:
```suggestion
    constexpr int bigTickSize = 10;
```

Reply:
should it be `static`?

Reply:
no, it's constexpr don't need static.

Reply:
Are you sure? They are apparently independent (ie constexpr does not imply static): [reference](https://stackoverflow.com/a/13867690)

Reply:
I was wrong. :) I tougth that constexpr variables where not created and simply translated to values during compilation. now I see that there's also consteval. I wish cpp gets simpler over time instead of creating more ways to declare values...